packages/flutter_tools/lib/src/crash_reporting.dart

```diff
@@ -0,0 +1,383 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+import 'package:stack_trace/stack_trace.dart';
+
+import 'globals.dart';
+import 'usage.dart';
+import 'version.dart';
+
+/// Tells crash backend that the error is from the Flutter CLI.
+String _kProductId = 'Flutter_Tools';
+
+/// Tells crash backend that this is a Dart error as opposed to, say, Java.
+String _kDartTypeId = 'DartError';
+
+/// Crash backend host.
+String _kCrashServerHost = 'clients2.google.com';
+
+/// Path to the crash servlet.
+// TODO(yjbanov): switch to non-staging when production is ready.
+String _kCrashEndpointPath = '/cr/staging_report';
+
+/// The field corresponding to the multipart/form-data file attachment where
+/// crash backend expects to find the Dart stack trace.
+String _kStackTraceFileField = 'DartError';
+
+/// The name of the file attached as [_kStackTraceFileField].
+///
+/// The precise value is not important. It is ignored by the crash back end, but
+/// it must be supplied in the request.
+String _kStackTraceFilename = 'stacktrace_file';
+
+bool _testing = false;
+
+/// Sends crash reports to Google.
+class CrashReportSender {
+  static final Uri _defaultBaseUri = new Uri(
+      scheme: 'https',
+      host: _kCrashServerHost,
+      port: 443,
+      path: _kCrashEndpointPath,
+  );
+
+  static Uri _baseUri = _defaultBaseUri;
+  final Usage _usage = Usage.instance;
+
+  /// Sends one crash report.
+  ///
+  /// The report is populated from data in [error] and [stackTrace].
+  Future<Null> sendReport({@required dynamic error, @required dynamic stackTrace}) async {
```

Comment:
Here I prefer named parameters because the types are the same, which makes it easy to call `sendReport(stack, error)` with no static or runtime errors. The only difference between the two is we send `error.runtimeType` vs `stackTrace.toString()`.